rpc/src/eth.rs

```diff
@@ -228,6 +241,86 @@ impl<B, C, P, CT, BE, H: ExHashT> EthApi<B, C, P, CT, BE, H> where
 		}
 		Ok(None)
 	}
+
+	fn current_block(&self, id: &BlockId<B>) -> Option<ethereum::Block> {
+		self.query_storage::<ethereum::Block>(
+			id,
+			&StorageKey(
+				storage_prefix_build(b"Ethereum", b"CurrentBlock")
+			)
+		)
+	}
+
+	fn current_statuses(&self, id: &BlockId<B>) -> Option<Vec<TransactionStatus>> {
+		self.query_storage::<Vec<TransactionStatus>>(
+			id,
+			&StorageKey(
+				storage_prefix_build(b"Ethereum", b"CurrentTransactionStatuses")
+			)
+		)
+	}
+
+	fn current_receipts(&self, id: &BlockId<B>) -> Option<Vec<ethereum::Receipt>> {
+		self.query_storage::<Vec<ethereum::Receipt>>(
+			id,
+			&StorageKey(
+				storage_prefix_build(b"Ethereum", b"CurrentReceipts")
+			)
+		)
+	}
+
+	fn account_codes(&self, id: &BlockId<B>, address: H160) -> Option<Vec<u8>> {
+		let mut key: Vec<u8> = storage_prefix_build(b"EVM", b"AccountCodes");
+		key.extend(blake2_128_extend(address.as_bytes()));
+		self.query_storage::<Vec<u8>>(
+			id,
+			&StorageKey(key)
+		)
+	}
+
+	fn account_storages(&self, id: &BlockId<B>, address: H160, index: U256) -> Option<H256> {
+		let tmp: &mut [u8; 32] = &mut [0; 32];
+		index.to_little_endian(tmp);
```

Comment:
@sorpaas please confirm this

Reply:
You mean confirm the endianness right? We should have a test to make sure we get the right data from storage.